### PR TITLE
Fix max_seq_length backward compatibility regression

### LINKED ISSUE #3910: max_length causing training loss issues not found in v0.19.1
### Reproduction

**Problem description**

When training using trl versions 0.20.0 and above and using max_length, I seem to get into a situation where the total loss during fine-tuning training is stagnant (image attached below. However, when using max_seq_length, training loss is decreasing as normal.

Doing some research, #2895 seems to have a relation to the variable name change, but it is only that, a variable name change, unless a max_seq_length was missed somewhere or max_length not added somewhere.

Please also note, I did test trl 0.21.0 to over 1000 epochs to see if it was an error on my end, however I still obtained stagnant results, which are completely the opposite at 10 epochs for version 0.19.1.

**Versions**
TRL: 0.19.1 (The one that works), =>0.20.0 ( The one that has issues)

**Configuration**

```python
sft_config = SFTConfig(
    ## GROUP 1: Memory usage
    # These arguments will squeeze the most out of your GPU's RAM
    # Checkpointing
    gradient_checkpointing=True,    # this saves a LOT of memory
    # Set this to avoid exceptions in newer versions of PyTorch
    gradient_checkpointing_kwargs={'use_reentrant': False}, 
    # Gradient Accumulation / Batch size
    # Actual batch (for updating) is same (1x) as micro-batch size
    gradient_accumulation_steps=1,  
    # The initial (micro) batch size to start off with
    per_device_train_batch_size=16, 
    # If batch size would cause OOM, halves its size until it works
    auto_find_batch_size=True,

    ## GROUP 2: Dataset-related
    max_seq_length=64,
    # Dataset
    # packing a dataset means no padding is needed
    packing=True,

    ## GROUP 3: These are typical training parameters
    num_train_epochs=10,
    learning_rate=3e-4,
    # Optimizer
    # 8-bit Adam optimizer - doesn't help much if you're using LoRA!
    optim='paged_adamw_8bit',       
    
    ## GROUP 4: Logging parameters
    logging_steps=10,
    logging_dir='./logs',
    output_dir='./llms/best-friend-llm',
    report_to='none'
)
```

**Outputs**

Before changing to max_length:

Image

Output after changing to max_length:

Image

### System Info

- Platform: Windows-10-10.0.19045-SP0
- Python version: 3.13.5
- TRL version: 0.19.1
- PyTorch version: 2.8.0+cu128
- accelerator(s): NVIDIA GeForce RTX 3060
- Transformers version: 4.55.2
- Accelerate version: 1.10.0
- Accelerate config: not found
- Datasets version: 4.0.0
- HF Hub version: 0.34.4
- bitsandbytes version: 0.47.0
- DeepSpeed version: not installed
- Diffusers version: not installed
- Liger-Kernel version: not installed
- LLM-Blender version: not installed
- OpenAI version: not installed
- PEFT version: 0.17.0
- vLLM version: not installed

### Checklist

- [x] I have checked that my issue isn't already filed (see [open issues](https://github.com/huggingface/trl/issues?q=is%3Aissue))
- [x] I have included my system information
- [x] Any code provided is minimal, complete, and reproducible ([more on MREs](https://docs.github.com/en/get-started/writing-on-github/working-with-advanced-formatting/creating-and-highlighting-code-blocks))
- [x] Any code provided is properly formatted in code blocks, (no screenshot, [more on code blocks](https://docs.github.com/en/get-started/writing-on-github/working-with-advanced-formatting/creating-and-highlighting-code-blocks))
- [x] Any traceback provided is complete

### PR DESCRIPTION
**Pull request by chi2liu**
Fixes #3910

## What

Restores backward compatibility for the deprecated `max_seq_length` parameter in SFTConfig.

## Why

- `max_seq_length` was renamed to `max_length` in v0.20.0
- The deprecation handler was removed, breaking existing code
- Users upgrading from v0.19.1 encounter errors instead of warnings

## How

- Added decorator to intercept the deprecated parameter
- Issues FutureWarning when `max_seq_length` is used
- Maps `max_seq_length` to `max_length` automatically
- Handles conflicts when both parameters are provided

## Testing

Added tests to verify:
- `max_seq_length` works with deprecation warning
- `max_length` works without warnings
- Proper handling when both parameters are provided

### CODE REVIEW

**Comment by qgallouedec** [comment]
`max_seq_length` was deprecated in 0.16 and removed in 0.20, we won't restore this parameter

**Comment by qgallouedec** [comment]
> Users upgrading from v0.19.1 encounter errors instead of warnings

Yes that's precisely what a deprecation warning is used for: warn user that using this param won't be possible in the future.

The fix is pretty easy though: just to rename this to `max_length` in your code
